shared/data/src/commonMain/kotlin/com/simplecityapps/shuttle/model/Song.kt

```diff
@@ -71,4 +71,19 @@ data class Song(
     } else {
         null
     }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || this::class != other::class) return false
+
+        other as Song
+
+        if (id != other.id) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return id.hashCode()
+    }
```

Comment:
I don't think it's a good idea to override the default behaviour of equals/hashcode here, as a convenience for comparing Song id's.

1.) This means that `Song` no longer behaves like a regular `data class`, which could be confusing / lead to unexpected behaviour
2.) I'm not sure if we're relying on the existing equality contract in other places, which could be broken by this change
3.) It's easy to just compare `song.id == otherSong.id`

Reply:
I was a bit overwhelmed when a wrote that, so I don't think I noticed it was a data class. Anyway, I didn't remember about the data classes having already a proper `equals` ;-)

Fixed in 3a82613.